packages/echo/src/client.ts

```diff
@@ -57,33 +57,75 @@ JSONSchemaFaker.option({
   alwaysFakeOptionals: true,
 });
 
-export class Echo {
+export class Client {
   private discoveredWorkflows: Array<DiscoverWorkflowOutput> = [];
 
   private ajv: Ajv;
 
   private templateEngine = new Liquid();
 
-  private readonly backendUrl?: string;
+  private readonly apiUrl: string;
 
   public apiKey?: string;
 
   public version: string = VERSION;
 
-  public devModeBypassAuthentication: boolean;
+  public strictAuthentication: boolean;
 
   public static NOVU_SIGNATURE_HEADER = HttpHeaderKeysEnum.SIGNATURE;
 
-  constructor(config?: ClientConfig) {
-    this.apiKey = config?.apiKey;
-    this.backendUrl = config?.backendUrl ?? DEFAULT_NOVU_API_BASE_URL;
-    this.devModeBypassAuthentication = config?.devModeBypassAuthentication || false;
+  constructor(options?: ClientOptions) {
+    const builtOpts = this.buildOptions(options);
+    this.apiKey = builtOpts.apiKey;
+    this.apiUrl = builtOpts.apiUrl;
+    this.strictAuthentication = builtOpts.strictAuthentication;
 
     const ajv = new Ajv({ useDefaults: true });
     addFormats(ajv);
     this.ajv = ajv;
   }
 
+  private buildOptions(providedOptions?: ClientOptions) {
+    const builtConfiguration: { apiKey?: string; apiUrl: string; strictAuthentication: boolean } = {
+      apiKey: undefined,
+      apiUrl: DEFAULT_NOVU_API_BASE_URL,
+      strictAuthentication: true,
+    };
+
+    // boolean flag risks getting parsed wrong in Object.entries, so this is to compensate
+    if (providedOptions && typeof providedOptions.strictAuthentication === 'boolean') {
+      builtConfiguration.strictAuthentication = providedOptions.strictAuthentication;
+    } else if (process.env.NOVU_STRICT_AUTHENTICATION === 'false') {
+      builtConfiguration.strictAuthentication = false;
+    }
+
+    Object.entries(builtConfiguration).reduce((_acc, [prop]) => {
+      if (prop === 'strictAuthentication') {
+        return '';
+      }
+      const envVar = `NOVU_${toConstantCase(prop)}`;
+      if (providedOptions && providedOptions[prop]) {
+        builtConfiguration[prop] = providedOptions[prop];
+      } else if (process.env[envVar]) {
+        builtConfiguration[prop] = process.env[envVar];
+      }
+
+      return '';
+    }, '');
+
+    return builtConfiguration;
+  }
```

Comment:
The client is getting a bit bloated with "useless" methods, methods that don't actually need to access instance properties (i.e. `this.<property>`) and can therefore be refactored to stateless functions.

I don't think we should action now, but let's keep this in mind for future refactoring - we should be able to reduce the size of the `Client` quite significantly.